engine/runtime/src/main/java/org/enso/interpreter/node/expression/builtin/meta/EqualsAnyNode.java

```diff
@@ -0,0 +1,609 @@
+      @CachedLibrary("otherTime") InteropLibrary otherInterop) {
+    LocalTime selfJavaTime;
+    LocalTime otherJavaTime;
+    try {
+      selfJavaTime = selfInterop.asTime(selfTime);
+      otherJavaTime = otherInterop.asTime(otherTime);
+    } catch (UnsupportedMessageException e) {
+      throw new IllegalStateException(e);
+    }
+    return selfJavaTime.equals(otherJavaTime);
+  }
+
+  @Specialization(guards = {
+      "selfInterop.isDuration(selfDuration)",
+      "otherInterop.isDuration(otherDuration)"
+  }, limit = "3")
+  boolean equalsDuration(Object selfDuration, Object otherDuration,
+      @CachedLibrary("selfDuration") InteropLibrary selfInterop,
+      @CachedLibrary("otherDuration") InteropLibrary otherInterop) {
+    Duration selfJavaDuration;
+    Duration otherJavaDuration;
+    try {
+      selfJavaDuration = selfInterop.asDuration(selfDuration);
+      otherJavaDuration = otherInterop.asDuration(otherDuration);
+    } catch (UnsupportedMessageException e) {
+      throw new IllegalStateException(e);
+    }
+    return selfJavaDuration.equals(otherJavaDuration);
+  }
+
+  /**
+   * Compares interop strings according to the lexicographical order, handling Unicode
+   * normalization. See {@code Text_Utils.compare_to}.
+   */
+  @TruffleBoundary
+  @Specialization(guards = {
+      "selfInterop.isString(selfString)",
+      "otherInterop.isString(otherString)"
+  }, limit = "3")
+  boolean equalsStrings(Object selfString, Object otherString,
+      @CachedLibrary("selfString") InteropLibrary selfInterop,
+      @CachedLibrary("otherString") InteropLibrary otherInterop) {
+    String selfJavaString;
+    String otherJavaString;
+    try {
+      selfJavaString = selfInterop.asString(selfString);
+      otherJavaString = otherInterop.asString(otherString);
+    } catch (UnsupportedMessageException e) {
+      throw new IllegalStateException(e);
+    }
+    return Normalizer.compare(
+        selfJavaString,
+        otherJavaString,
+        Normalizer.FOLD_CASE_DEFAULT
+    ) == 0;
+  }
+
+  @Specialization(guards = {
+      "selfInterop.isExecutable(selfFunction)",
+      "otherInterop.isExecutable(otherFunction)",
+      "selfInterop.isMemberInvocable(selfFunction, EQUALS_MEMBER_NAME)",
+      "otherInterop.isMemberInvocable(otherFunction, EQUALS_MEMBER_NAME)",
+  }, limit = "3")
+  boolean equalsFunctions(Object selfFunction, Object otherFunction,
+      @CachedLibrary("selfFunction") InteropLibrary selfInterop,
+      @CachedLibrary("otherFunction") InteropLibrary otherInterop,
+      @CachedLibrary(limit = "3") InteropLibrary retValueInterop) {
+    Object ret;
+    try {
+      ret = selfInterop.invokeMember(selfFunction, MethodNames.Function.EQUALS, otherFunction);
+      return retValueInterop.asBoolean(ret);
+    } catch (UnsupportedMessageException | ArityException | UnknownIdentifierException |
+             UnsupportedTypeException e) {
+      throw new IllegalStateException(e);
+    }
+  }
```

Comment:
What is the case handled by this specialization? Do we have a test for it?

Reply:
I tried to provide the same functionality as in [Function.java:210](), and [Function.scala:27](), which seems to provide a notion for equality of two `Function`. But it seems so far that this is a dead code. I will try to remove this specialization and see if everything works without it.

Reply:
Ok

Reply:
I tried removing, and everything still works. Removing permanently.